packages/auth/src/grant/routes.ts

```diff
@@ -206,13 +266,20 @@ async function continueGrant(
     interactionService
   } = deps
 
-  const interaction = await interactionService.getByRef(interactRef)
+  if (!ctx.request.body || Object.keys(ctx.request.body).length === 0) {
+    await pollGrantContinuation(deps, ctx, continueId, continueToken)
+    return
+  }
+
+  const interaction = await interactionService.getByRef(interactRef as string)
   if (
     !interaction ||
     !isContinuableGrant(interaction.grant) ||
     !isMatchingContinueRequest(continueId, continueToken, interaction.grant)
   ) {
     ctx.throw(404, { error: 'unknown_request' })
+  } else if (isGrantStillWaiting(interaction.grant)) {
+    ctx.throw(401, { error: 'too_fast' })
```

Comment:
I see this [here](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol-15#name-continuing-a-grant-request):
> If the client instance does not respect the given wait period, the AS MUST return the too_fast error ([Section 3.6](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol-15#response-error)).

More of a question about the spec, but is it an actual issue if the continuation request (somehow) is made before the `wait` time with the correct `interact_ref`?

Reply:
I interpreted this as meaning that the wait time should be respected irregardless if the request is made with the correct `interact_ref`.

Reply:
Makes sense, I doubt that is something that would realistically happen anyway